File: tests/rustdoc-js/enum-variant-not-type.js
const EXPECTED = [
    {
        'query': 'T -> T',
        'correction': null,
        'others': [
            {
                'path': 'enum_variant_not_type',
                'name': 'my_fn',
            },
            {
                'path': 'enum_variant_not_type::AutoCorrectConfounder',
                'name': 'assoc_type_acts_like_generic',
            },
        ],
    },
    {
        'query': 'InsertUnnecessarilyLongTypeNameHere -> InsertUnnecessarilyLongTypeNameHere',
        'correction': null,
        'others': [
            {
                'path': 'enum_variant_not_type',
                'name': 'my_fn',
            },
            {
                'path': 'enum_variant_not_type::AutoCorrectConfounder',
                'name': 'assoc_type_acts_like_generic',
            },
        ],
    },
    {
        'query': 'InsertUnnecessarilyLongTypeNameHere',
        'correction': null,
        'others': [
            {
                'path': 'enum_variant_not_type::AutoCorrectConfounder',
                'name': 'InsertUnnecessarilyLongTypeNameHere',
            },
        ],
    },
    {
        'query': 'InsertUnnecessarilyLongTypeNameHereX',
        'correction': null,
        'others': [
            {
                'path': 'enum_variant_not_type::AutoCorrectConfounder',
                'name': 'InsertUnnecessarilyLongTypeNameHere',
            },
        ],
    },
    {
        'query': 'T',
        'correction': null,
        'others': [
            {
                'path': 'enum_variant_not_type::MyTrait',
                'name': 'T',
            },
        ],
    },
    {
        'query': 'T',
        'correction': null,
        'others': [
            {
                'path': 'enum_variant_not_type::MyTrait',
                'name': 'T',
            },
        ],
    },
];

File: tests/rustdoc-js/enum-variant-not-type.rs
pub trait MyTrait {
    // Reduced from `arti` crate.
    // https://tpo.pages.torproject.net/core/doc/rust/tor_config/list_builder/trait.DirectDefaultEmptyListBuilderAccessors.html#associatedtype.T
    type T;
    fn not_appearing(&self) -> Option<&Self::T>;
}

pub fn my_fn<X>(t: X) -> X {
    t
}

pub trait AutoCorrectConfounder {
    type InsertUnnecessarilyLongTypeNameHere;
    fn assoc_type_acts_like_generic(
        &self,
        x: &Self::InsertUnnecessarilyLongTypeNameHere,
    ) -> Option<&Self::InsertUnnecessarilyLongTypeNameHere>;
}
